core/src/main/scala/org/apache/spark/shuffle/ShuffleChecksumUtils.scala

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.shuffle
+
+import java.io.{DataInputStream, File, FileInputStream}
+import java.util.zip.CheckedInputStream
+
+import org.apache.spark.network.shuffle.checksum.ShuffleChecksumHelper
+import org.apache.spark.network.util.LimitedInputStream
+
+object ShuffleChecksumUtils {
+
+  /**
+   * Ensure that the checksum values are consistent with index file and data file.
+   */
+  def compareChecksums(
```

Comment:
Should we put this into `ShuffleChecksumHelper`?

Reply:
`ShuffleChecksumHelper` is `test` code helper under `test/scala/...`. So we can not reuse it there.

Reply:
Hmm, I meant `ShuffleChecksumHelper` not `ShuffleChecksumTestHelper`.

Reply:
Oh, I misread that.

BTW, I also considered that but I decided not to do because it's `Java` class which means we re-implemented the whole logic. I guess you also suggested a moving from `Scala` to `Scala` code, not `Scala` to `Java`. Or, did you suggest to reimplement with `Java`?

Reply:
Ah I see. No, I didn't realized that `ShuffleChecksumHelper` is Java code. Thanks for reply.